storage/redis_cluster.go

```diff
@@ -375,6 +361,23 @@ func (r *RedisCluster) SetRawKey(keyName, session string, timeout int64) error {
 	return nil
 }
 
+func (r *RedisCluster) SetNX(keyName string, value interface{}, timeout int64) (bool, error) {
```

Comment:
Why are we allowing `interface{}` here?

Reply:
not to restrict the value, it' the same in redis client

Reply:
The redis client is generic, but we are not, you don't need it to be an interface? As it's intended purpose is a shared-lock mechanism, could we just name it `Lock(key string, timeout int64) (bool, error)`?